core/src/components/datetime/datetime.tsx

```diff
@@ -959,6 +968,21 @@ export class Datetime implements ComponentInterface {
     root.addEventListener('ionBlur', (ev: Event) => ev.stopPropagation());
   }
 
+  /**
+   * When the presentation is changed, all calendar content is recreated,
+   * so we need to re-init behavior with the new elements.
+   */
+  componentDidRender() {
+    if (this.prevPresentation === null) {
+      this.prevPresentation = this.presentation;
+    }
+
+    if (this.presentation === this.prevPresentation) { return; }
```

Comment:
Can we cache the values of prevPresentation and presentation? Every time we access `this.[variable]` we incur a small perf hit in Stencil, so caching the value lets us avoid those extra hits.

Also can we return in the `null` check? If we make it into the body of the block, it's just going to return on line 980 anyways, so better to avoid the additional equality check.

```js
const { prevPresentation, presentation } = this;

if (prevPresentation === null) {
  this.prevPresentation = presentation; 
  return;
}

if (prevPresentation === presentation) { return; }

this.prevPresentation = presentation;

this.destroyListeners();
this.initializeListeners();
```